src/sentry/models/widget.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import absolute_import, print_function
+
+from django.db import models
+from django.utils import timezone
+from jsonfield import JSONField
+
+from sentry.constants import ObjectStatus
+from sentry.db.models import BoundedPositiveIntegerField, FlexibleForeignKey, Model, sane_repr
+
+
+class WidgetDisplayTypes(object):
+    LINE_CHART = 0
+    AREA_CHART = 1
+    STACKED_AREA_CHART = 2
+    BAR_CHART = 3
+    PIE_CHART = 4
+    TABLE = 5
+    WORLD_MAP = 6
+    PERCENTAGE_AREA_CHART = 7
+
+    @classmethod
+    def as_choices(cls):
+        return [
+            (cls.LINE_CHART, 'line'),
+            (cls.AREA_CHART, 'area'),
+            (cls.STACKED_AREA_CHART, 'stacked_area'),
+            (cls.BAR_CHART, 'bar'),
+            (cls.PIE_CHART, 'pie'),
+            (cls.TABLE, 'table'),
+            (cls.WORLD_MAP, 'world_map'),
+            (cls.PERCENTAGE_AREA_CHART, 'percentage_area_chart')
+        ]
+
+
+class WidgetDataSourceTypes(object):
+    DISCOVER_SAVED_SEARCH = 0
+    API_QUERY = 1  # ehhhh.....
```

Comment:
Do we need this? What would dashboards look like if all the datasources were saved queries? Does that remove any must have features for the first release?

Reply:
Yeah we can remove api as a source right now, there are no plans for API as source for the initial release.